tests/hotplug/cpu.go

```diff
@@ -0,0 +1,194 @@
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libwait"
+)
+
+var _ = Describe("[sig-compute][serial]CPU Hotplug", decorators.SigCompute, decorators.RequiresTwoSchedulableNodes, func() {
+	var virtClient kubecli.KubevirtClient
+	var originalKv *v1.KubeVirt
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+		originalKv = util2.GetCurrentKv(virtClient)
+		updateStrategy := &v1.KubeVirtWorkloadUpdateStrategy{
+			WorkloadUpdateMethods: []v1.WorkloadUpdateMethod{v1.WorkloadUpdateMethodLiveMigrate},
+		}
+		patchWorkloadUpdateMethod(originalKv.Name, virtClient, updateStrategy)
+
+	})
+
+	AfterEach(func() {
+		patchWorkloadUpdateMethod(originalKv.Name, virtClient, &originalKv.Spec.WorkloadUpdateStrategy)
+	})
+
+	Context("A VM with cpu.maxSockets set higher than cpu.sockets", func() {
+		type cpuCount struct {
+			enabled  int
+			disabled int
+		}
+		countDomCPUs := func(vmi *v1.VirtualMachineInstance) (count cpuCount) {
+			domSpec, err := tests.GetRunningVMIDomainSpec(vmi)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(domSpec.VCPUs).NotTo(BeNil())
+			for _, cpu := range domSpec.VCPUs.VCPU {
+				if cpu.Enabled == "yes" {
+					count.enabled++
+				} else {
+					ExpectWithOffset(1, cpu.Enabled).To(Equal("no"))
+					ExpectWithOffset(1, cpu.Hotpluggable).To(Equal("yes"))
+					count.disabled++
+				}
+			}
+			return
+		}
+		It("should have spare CPUs that be enabled and more pod CPU resource after a migration", func() {
+			By("Creating a running VM with 1 socket and 2 max sockets")
+			const (
+				maxSockets uint32 = 2
+			)
+
+			vmi := libvmi.NewAlpine()
+			vmi.Spec.Domain.CPU = &v1.CPU{
+				Cores:   2,
+				Sockets: 1,
+				Threads: 1,
+			}
+			vm := tests.NewRandomVirtualMachine(vmi, true)
+			vm.Spec.LiveUpdateFeatures = &v1.LiveUpdateFeatures{
+				CPU: &v1.LiveUpdateCPU{
+					MaxSockets: pointer.P(maxSockets),
+				},
+			}
+
+			vm, err := virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(func() error {
+				vmi, err = virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+				return err
+			}).ShouldNot(HaveOccurred())
+			libwait.WaitForSuccessfulVMIStart(vmi)
+
+			By("Ensuring the computee container has 200m CPU")
+			pod := tests.GetVmiPod(virtClient, vmi)
+			var compute *k8sv1.Container
+			for _, container := range pod.Spec.Containers {
+				if container.Name == "compute" {
+					compute = &container
+					break
+				}
+			}
```

Comment:
nit: better use [GetComputeContainerOfPod()](https://github.com/kubevirt/kubevirt/blob/v1.0.0-beta.0/tests/utils.go#L504) dedicated function for that